Replace manual cache in _python_dispatch.get_alias_info with functools.cache

swolchok opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #161466
* #161455
* #161438
* #161433
* #161432
* #161329
* #161328
* #161317
* #161315
* #161308
* #161304
* #161292
* #161301
* #161300
* __->__ #161286

In addition to being more code, the manual cache was doing an extra dictionary lookup on each cache hit.

swolchok committed:
Replace manual cache in _python_dispatch.get_alias_info with functools.cache

In addition to being more code, the manual cache was doing an extra dictionary lookup on each cache hit.

[ghstack-poisoned]

swolchok committed:
Update on "Replace manual cache in _python_dispatch.get_alias_info with functools.cache"


In addition to being more code, the manual cache was doing an extra dictionary lookup on each cache hit.

[ghstack-poisoned]

swolchok committed:
Update on "Replace manual cache in _python_dispatch.get_alias_info with functools.cache"


In addition to being more code, the manual cache was doing an extra dictionary lookup on each cache hit.

[ghstack-poisoned]

wconstab commented on torch/utils/_python_dispatch.py:
```diff
@@ -527,15 +528,10 @@ class SchemaInfo:
-
 # Given an OpOverload, returns schema information on it.
 # This is cached for efficiency, since it can involve running torchgen
+@functools.cache
```
for my education- is this purely a "its more idiomatic this way" thing or is this faster too?

swolchok committed:
Update on "Replace manual cache in _python_dispatch.get_alias_info with functools.cache"


In addition to being more code, the manual cache was doing an extra dictionary lookup on each cache hit.

[ghstack-poisoned]

swolchok commented:
@pytorchbot merge
